src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -530,7 +533,7 @@ public void run() {
                 }
                 logger.warn("[{}] deleting dangling index", metaData.index());
                 try {
-                    indicesService.deleteIndexStore("deleting dangling index", metaData);
+                    indicesService.deleteIndexStore("deleting dangling index", metaData, clusterService.state());
```

Comment:
also, in IndicesClusterStateService, we pass a different IndexMetaData, so the flexibility to pass them both is needed?